codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -121,7 +121,7 @@ public void remoteSettings(Http2Settings settings) throws Http2Exception {
 
         Long maxConcurrentStreams = settings.maxConcurrentStreams();
         if (maxConcurrentStreams != null) {
-            connection.local().maxStreams((int) Math.min(maxConcurrentStreams, Integer.MAX_VALUE));
+            connection.local().maxActiveStreams((int) Math.min(maxConcurrentStreams, Integer.MAX_VALUE));
```

Comment:
@normanmaurer I'm assuming your suggestion is to directly cast the long to an int here? Wouldn't that allow us to get a negative value?